src/routes/transactions/mappers/common/swap-order.mapper.ts

```diff
@@ -0,0 +1,239 @@
+      return this._mapUnknownOrderStatus(chainId, transaction, dataSize);
+    }
+
+    const orderUid: `0x${string}` | null =
+      this.setPreSignatureDecoder.getOrderUid(transaction.data);
+    if (!orderUid) {
+      return this._mapUnknownOrderStatus(chainId, transaction, dataSize);
+    }
+    const order = await this.swapsRepository.getOrder(chainId, orderUid);
+
+    try {
+      const [buyToken, sellToken] = await Promise.all([
+        this.tokenRepository.getToken({
+          chainId,
+          address: order.buyToken,
+        }),
+        this.tokenRepository.getToken({
+          chainId,
+          address: order.sellToken,
+        }),
+      ]);
+
+      const buyTokenAmount = new TokenAmount({
+        token: buyToken,
+        amount: order.buyAmount,
+        executedAmount: order.executedBuyAmount,
+      });
+      const sellTokenAmount = new TokenAmount({
+        token: sellToken,
+        amount: order.sellAmount,
+        executedAmount: order.executedSellAmount,
+      });
+
+      switch (order.status) {
+        case 'fulfilled':
+          return this._mapFulfilledOrderStatus({
+            buyToken: buyTokenAmount,
+            sellToken: sellTokenAmount,
+            order,
+          });
+        case 'open':
+        case 'cancelled':
+        case 'expired':
+          return this._mapDefaultOrderStatus({
+            buyToken: buyTokenAmount,
+            sellToken: sellTokenAmount,
+            order,
+          });
+        default:
+          return this._mapUnknownOrderStatus(chainId, transaction, dataSize);
+      }
+    } catch (e) {
+      this.loggingService.warn(e);
+      return this.customTransactionMapper.mapCustomTransaction(
+        transaction,
+        dataSize,
+        chainId,
+        null,
+        null,
+      );
+    }
+  }
+
+  private _mapUnknownOrderStatus(
+    chainId: string,
+    transaction: MultisigTransaction | ModuleTransaction,
+    dataSize: number,
+  ): Promise<CustomTransactionInfo> {
+    return this.customTransactionMapper.mapCustomTransaction(
+      transaction,
+      dataSize,
+      chainId,
+      null,
+      null,
+    );
+  }
+
+  private _getExecutionPrice(
+    sellToken: TokenAmount,
+    buyToken: TokenAmount,
+  ): string {
+    const ratio = sellToken.getExecutedAmount() / buyToken.getExecutedAmount();
+    return `1 ${sellToken.token.symbol} = ${ratio} ${buyToken.token.symbol}`;
+  }
+
+  private _getLimitPrice(
+    sellToken: TokenAmount,
+    buyToken: TokenAmount,
+  ): string {
+    const ratio = sellToken.getAmount() / buyToken.getAmount();
+    return `1 ${sellToken.token.symbol} = ${ratio} ${buyToken.token.symbol}`;
+  }
+
+  private _mapFulfilledOrderStatus(args: {
+    buyToken: TokenAmount;
+    sellToken: TokenAmount;
+    order: Order;
+  }): SwapOrderTransactionInfo {
+    if (args.order.kind === 'unknown') {
+      throw new Error('Unknown order kind');
+    }
+    const surplusFee: string | null = args.order.executedSurplusFee
```

Comment:
```suggestion
    const surplusFee = args.order.executedSurplusFee
```

Reply:
I've updated the names to be more explicit on what we expect of the value (since it's a label) – [b56ec3b](https://github.com/safe-global/safe-client-gateway/pull/1299/commits/b56ec3bbc1530ceb762c0a77d495bb99368bff3c)